gips/data/core.py

```diff
@@ -792,16 +792,16 @@ def products2assets(cls, products):
         return set(assets)
 
     @classmethod
-    def fetch(cls, products, tiles, textent):
-        """ Download data for tiles and add to archive """
+    def fetch(cls, products, tiles, textent, update=False):
+        """ Download data for tiles and add to archive. update forces fetch """
         assets = cls.products2assets(products)
         fetched = []
         for a in assets:
             for t in tiles:
                 asset_dates = cls.Asset.dates(a, t, textent.datebounds, textent.daybounds)
                 for d in asset_dates:
-                    # if we don't have it already
-                    if not cls.Asset.discover(t, d, a):
+                    # if we don't have it already, or if update (force) flag
+                    if not cls.Asset.discover(t, d, a) or update == True:
```

Comment:
Sure, that makes sense.

> On Jul 5, 2016, at 9:29 AM, ags-tolson <notifications@github.com> wrote:
> 
> In gips/data/core.py <https://github.com/Applied-GeoSolutions/gips/pull/53#discussion_r69562343>:
> 
> >          assets = cls.products2assets(products)
> >          fetched = []
> >          for a in assets:
> >              for t in tiles:
> >                  asset_dates = cls.Asset.dates(a, t, textent.datebounds, textent.daybounds)
> >                  for d in asset_dates:
> > -                    # if we don't have it already
> > -                    if not cls.Asset.discover(t, d, a):
> > +                    # if we don't have it already, or if update (force) flag
> > +                    if not cls.Asset.discover(t, d, a) or update == True:
> The or checks for truth so you don't need to do it explicitly:
> 
> if not cls.Asset.discover(t, d, a) or update:
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/Applied-GeoSolutions/gips/pull/53/files/3a9deaff37b928cc1fba52d1cdb2083f4a8127db#r69562343>, or mute the thread <https://github.com/notifications/unsubscribe/ADD4XNAayR-HTPltae9mEnFn9KB1XeX8ks5qSlw3gaJpZM4JDj03>.
>